### PR TITLE
248: More basic setup

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Add meta setup for IntelliJ, git and Github.

### Proposed changes

- Add PR and issue templates
- Add CODEOWNERS file to automatically request reviews
- Add [fvm](https://fvm.app/) configuration
- Add IntelliJ configuration
- Add IntelliJ run configurations to start the app in development and production
- Update .gitignore (adjust at https://www.toptal.com/developers/gitignore?templates=flutter,intellij,visualstudiocode)

### Side effects

None.

### Testing

None.

### Resolved issues

Fixes: #248.

---

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Each line is a file pattern followed by one or more owners.
+
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence,
+# these owners will be requested for
+# review when someone opens a pull request.
+*       @NikoHadouken @volkramweber @steffenkleinle @Stift
```
Comment (steffenkleinle) [inline CODEOWNERS:7]: @Stift I am a little confused if you are the "right" Christian and whether you will work on the flutter app? :D 
If you are, do you want to get review notifications?
If you are not, could you perhaps answer with the correct Github handle of the other Christian? Thank you :)

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (steffenkleinle) [inline .github/ISSUE_TEMPLATE/bug-report.md:1]: I just copied all these templates, if you'd like to change something, please let me know.

File: .fvmrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "flutter": "3.24.3"
```
Comment (steffenkleinle) [inline .fvmrc:2]: I can really recommend using fvm for flutter/dart version management, especially if you are developing on multiple flutter projects (with different flutter versions) at the same time.

**Comment by volkramweber** [comment]
@steffenkleinle Sure to commit IntelliJ configs? I've never seen this before, what are the advantages?

**Comment by steffenkleinle** [comment]
> @steffenkleinle Sure to commit IntelliJ configs? I've never seen this before, what are the advantages?

We actually do this in all our projects, but I am open for just ignoring the whole folder. I think advantages are:
- Settings that should be the same among all devs are the same (e.g. line width in this case)
- You can share settings and configuration that is nice to have (e.g. the run configurations in this case)

If you have the feeling that some of these are not correct or differ for you, we can also just add those files to .gitignore (also at a later point if you realize problems). There are also a few files ignored already (e.g. workspace.xml).

**Comment by volkramweber** [comment]
> > @steffenkleinle Sure to commit IntelliJ configs? I've never seen this before, what are the advantages?
> 
> We actually do this in all our projects, but I am open for just ignoring the whole folder. I think advantages are:
> 
> * Settings that should be the same among all devs are the same (e.g. line width in this case)
> * You can share settings and configuration that is nice to have (e.g. the run configurations in this case)
> 
> If you have the feeling that some of these are not correct or differ for you, we can also just add those files to .gitignore (also at a later point if you realize problems). There are also a few files ignored already (e.g. workspace.xml).

Sounds good!

File: .gitignore
Comment (volkramweber) [inline .gitignore:1]: Shouldn't we add `.DS_Store` to this file to prevent other developers from accidentally commit these files?

Comment (steffenkleinle) [reply .gitignore:1]: Sure, we can do that.

**Review by volkramweber — APPROVED** [review]

**Pull request merged**